qiskit/transpiler/passes/optimization/normalize_rx_angle.py

```diff
@@ -0,0 +1,149 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Performs three optimizations to reduce the number of pulse calibrations for
+the single-pulse RX gates:
+Wrap RX Gate rotation angles into [0, pi] by sandwiching them with RZ gates.
+Convert RX(pi/2) to SX, and RX(pi) to X if the calibrations exist in the target.
+Quantize the RX rotation angles using a resolution provided by the user.
+"""
+
+import numpy as np
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.circuit.library.standard_gates import RXGate, RZGate, SXGate, XGate
+
+
+class NormalizeRXAngle(TransformationPass):
+    """Wrap RX Gate rotation angles into [0, pi] by sandwiching them with RZ gates.
+    This will help reduce the size of calibration data,
+    as we won't have to keep separate, phase-flipped calibrations for negative rotation angles.
+    Moreover, if the calibrations exist in the target, convert RX(pi/2) to SX, and RX(pi) to X.
+    This will allow us to exploit the more accurate, hardware-calibrated pulses.
+    Lastly, quantize the RX rotation angles using a resolution provided by the user.
+    """
```

Comment:
```suggestion
    """Normalize theta parameter of RXGate instruction.
    
    The parameter normalization is performed with following steps.
    
    1) Wrap RX Gate theta into [0, pi]. When theta is negative value, the gate is 
    decomposed into the following sequence.
    
    .. code-block::
    
       ┌───────┐┌─────────┐┌────────┐
    q: ┤ Rz(π) ├┤ Rx(|θ|) ├┤ Rz(-π) ├
       └───────┘└─────────┘└────────┘
    
    2) If the operation is supported by target, convert RX(pi/2) to SX, and RX(pi) to X.
    
    3) Quantize theta value according to the user-specified resolution.
    
    This will help reduce the size of calibration data sent over the wire,
    and allow us to exploit the more accurate, hardware-calibrated pulses.
    Note that pulse calibration might be attached per each rotation angle.
    """
```

First line must be short followed by blank line, since this line shows up in the auto summary. For example:
![image](https://github.com/Qiskit/qiskit/assets/39517270/866c8dd6-4c32-4c57-90cc-d705c98e8b1b)

Reply:
Thank you for the suggestion and the explanation!